tests/cpp_unit_tests/test_transformer_tap_regulator.cpp

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#include <power_grid_model/component/transformer_tap_regulator.hpp>
+
+#include <doctest/doctest.h>
+
+namespace power_grid_model {
+
+TEST_CASE("Test transformer tap regulator") {
+    TransformerTapRegulatorInput const input{.id = 1,
+                                             .regulated_object = 2,
+                                             .control_side = ControlSide::from,
+                                             .u_set = 10.0e3,
+                                             .u_band = 1.0e3,
+                                             .status = true,
+                                             .line_drop_compensation_r = 1.0,
+                                             .line_drop_compensation_x = 2.0};
```

Comment:
`regulated_object` and `control_side` don't seem to appear in the definition if  `TransformerTapRegulatorInput`
```suggestion
TEST_CASE("Test transformer tap regulator") {
    TransformerTapRegulatorInput const input{.id = 1,
                                             .status = true,
                                             .u_set = 10.0e3,
                                             .u_band = 1.0e3,
                                             //.regulated_object = 2,
                                             //.control_side = ControlSide::from,
                                             .line_drop_compensation_r = 1.0,
                                             .line_drop_compensation_x = 2.0};
```

Reply:
That's not correct, they are in `TransformerTapRegulatorInput` ;)